pkg/sdk/action.go

```diff
@@ -54,6 +56,43 @@ func Create(object Object) (err error) {
 	return nil
 }
 
+// Patches provided object on the server with given patch and updates the arg
+// "object" with the result from the server(UID, resourceVersion, etc).
+// Returns an error if the object’s TypeMeta(Kind, APIVersion) or ObjectMeta(Name, Namespace) is missing or incorrect.
+// Returns an error if patch couldn't be json serialized into bytes.
+// Can also return an api error from the server
+// e.g Conflict https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L428
+func Patch(patch interface{}, object Object, pt types.PatchType) (err error) {
```

Comment:
@himanshug Can you give an example of how you construct the patch object and call this function?
I mainly want to see how the first argument `patch` would be constructed.
I thought patches were already constructed as byte arrays.
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/jsonmergepatch/patch.go#L31

Why is the type `interface{}`?

Reply:
thanks for taking a look

In my current code, at the call site it looks like
```
sdk.Patch(map[string]v1alpha1.ClusterStatus{"status": updatedStatus}, baseObj, types.MergePatchType)
```

where `ClusterStatus` is a custom struct type in my code and is part of my k8s CRD for the operator. `patch` needs to be "something" (a map, a struct or whatever) that can be serialized into `[]byte` and resource kind represented by `object` can be patched with that. ( you might have seen https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/ this already, but that has example of what patches look like)
`interface{}` is golang's way of saying I accept anything which is appropriate for the arg `patch`.

that said, as I write this, I think , we could take "[]byte patch" instead in the api as is done for `ResourceInterface.Patch(name string, pt types.PatchType, data []byte) (*unstructured.Unstructured, error)` and user would be responsible for creating the []byte (and to allow use cases where user already has the []byte for some reason for the patch rather than a struct, map etc)

let me know if that sounds good, I will make the change.

Reply:
@himanshug Yes I think changing the patch arg type to `[]byte` is better since it's closer to `ResourceInterface.Patch(name string, pt types.PatchType, data []byte)`.
Plus the apimachinery utils for constructing patches also return byte arrays.

Reply:
changed.